src/signals.jl

```diff
@@ -129,6 +129,53 @@ Invoke/return `Legolas.write(path_or_io, signals, Schema("onda.signal@1"); kwarg
 """
 write_signals(path_or_io, signals; kwargs...) = Legolas.write(path_or_io, signals, Legolas.Schema("onda.signal@1"); kwargs...)
 
+"""
+    validate_signals(signals)
+
+Perform both table-level and row-level validation checks on the content of `signals`,
+a presumed `onda.signal` table. Returns `signals`.
+
+This function will throw an error in any of the following cases:
+
+- `Legolas.validate(signals, Legolas.Schema("onda.signal@1"))` throws an error
+- `Signal(row)` errors for any `row in Tables.rows(signals)`
+- `signals` contains rows with duplicate `file_path`s
+- `signals` contains rows that share a `recording` and `kind` but have overlapping `span`s
```

Comment:
this is going to cause problems for recordings where the source is malformed in some way, e.g. an EDF file with EMG signals that have different sampling rates and hence can't be combined into a single `Samples` without some kind of lossy transformation

Reply:
I'm unsure what you mean, can you give an example? I think there might either be a misunderstanding at play or you're referring to an OndaEDF behavior or something i'm unaware of

just to recap the relevant parts of the spec: each `onda.signal` row is the metadata that describes a single sample data blob. so if you have multiple sample data blobs, you should have multiple `signal`s. multiple `signals` with the same `kind` (side note: honestly at this point I wish we had kept this field named `name`) are considered to be discontiguous samples of the same underlying process, with the time range given by `span`

Reply:
Oh shoot the spec explicitly allows overlap:

> ...thus enabling the representation/storage of discontiguous/**overlapping** sample data.

Hmm. I'm not sure we should allow that

Reply:
Discussed in Beacon slack. We actually have data for allowing overlap is useful. Turns out this a hidden feature that past Jarrett gifted to future Jarrett when he wrote `overlapping`. I'll remove this from the validation checks

Reply:
We have encountered EEG data where there are signals of the same "type" (for instance, EMG) but with different sampling rates and units (e.g. leg emg vs. chin emg).  so being able to represent overlapping signals of the same kind but different properties is important in practice.